File: 1013-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        fibonacci_nums = [0, 1]
        if n >= 2:
            for i in range(2, n+1):
                fibonacci_nums.append(fibonacci_nums[i-2]+fibonacci_nums[i-1])
        return fibonacci_nums[n]


        

File: 1013-fibonacci-number/fibonacci-number.go
import "fmt"

func fib(n int) int {
	if n == 0 {
		return 0
	} else if n == 1 {
        return 1
    }
	fibonacciNums := make([]int, n+1)
	fibonacciNums[1] = 1
	if n < 2 {
		return fibonacciNums[n]
	}
	for i := 2; i <= n; i++ {
		fmt.Println(i, fibonacciNums)
		fibonacciNums[i] = fibonacciNums[i-1] + fibonacciNums[i-2]
	}
	fmt.Println(fibonacciNums)
	return fibonacciNums[n]
}